src/negotiation/order.ts

```diff
@@ -71,20 +80,74 @@ export function orderSwapAssetMatchesForMaker(
 
 export function assetOrderToSwap(orderAsset: OrderAsset): Asset | undefined {
   if (isNative(orderAsset)) {
+    const name = orderAsset.asset;
+    const quantity = fromNominal(name, orderAsset.nominalAmount);
+    if (!quantity) {
+      return undefined;
+    }
     return {
-      name: orderAsset.asset,
-      quantity: orderAsset.amount
+      name,
+      quantity: quantity.toString()
     };
   }
 
   const token = getToken(orderAsset.asset);
   if (token) {
+    const name = token.type.toLowerCase();
+    const quantity = fromNominal(name, orderAsset.nominalAmount, token);
+    if (!quantity) {
+      return undefined;
+    }
+
     return {
-      name: token.type.toLowerCase(),
-      quantity: orderAsset.amount,
+      name,
+      quantity: quantity.toString(),
       token_contract: token.address
     };
   }
 
   return undefined;
 }
+
+function areAmountsEqual(
```

Comment:
:thinking: As a future improvement it could be nice to create a generic interface `Equals<T>` to be implemented on types such as `BitcoinAmount` and `EtherQuantity`.

Reply:
We don't have such types but I see your point :)